### PR TITLE
Validate database name and output error message to the user

### PR DESCRIPTION
**Pull request by Julien4218**

### CODE REVIEW

**Review by zlesnr — COMMENTED** [review]
We might also want to check that those variables are not empty to avoid the syntax error like we saw in the logs recipe.  What do you think?

**Comment by Julien4218** [comment]
I think the current addition is also validating that and will also fail with an error message to the user saying that database name is invalid

File: recipes/newrelic/infrastructure/ohi/mysql/debian.yml [hunk 1]
```diff
@@ -92,7 +92,13 @@ install:
...
-            echo -e "[Error]: Provided user has no access to the desired database.\n - See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mysql-monitoring-integration#config for more info."
+            echo "[Error]: Provided user has no access to the desired database. See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mysql-monitoring-integration#config for more info." >> /dev/stderr
+            exit 2
+          fi
+        - |
+          IS_DATABASE_VALID=$(mysql -u{{.NR_CLI_DB_USERNAME}} --port {{.NR_CLI_DB_PORT}} -p{{.NR_CLI_DB_PASSWORD}} -e "use {{.NR_CLI_DATABASE}};SELECT CURDATE()" 2>&1 | grep -v "mysql: [Warning] Using a password on the command line interface can be insecure." | grep 'CURDATE()' | grep -v grep | wc -l)
+          if [ $IS_DATABASE_VALID -eq 0  ]; then
+            echo "[Error]: The provided database name is not accessible with the provided MySQL username and password. See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mysql-monitoring-integration#config for more info." >> /dev/stderr
             exit 2
```
Comment (jbeveland27) [inline recipes/newrelic/infrastructure/ohi/mysql/debian.yml:102]: Should the exit code be incremented to `3` to be able to distinguish the type of error?

File: recipes/newrelic/infrastructure/ohi/mysql/rhel.yml [hunk 2]
```diff
@@ -92,7 +92,13 @@ install:
...
-            echo -e "[Error]: Provided user has no access to the desired database.\n - See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mysql-monitoring-integration#config for more info."
+            echo "[Error]: Provided user has no access to the desired database. See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mysql-monitoring-integration#config for more info." >> /dev/stderr
+            exit 2
+          fi
+        - |
+          IS_DATABASE_VALID=$(mysql -u{{.NR_CLI_DB_USERNAME}} --port {{.NR_CLI_DB_PORT}} -p{{.NR_CLI_DB_PASSWORD}} -e "use {{.NR_CLI_DATABASE}};SELECT CURDATE()" 2>&1 | grep -v "mysql: [Warning] Using a password on the command line interface can be insecure." | grep 'CURDATE()' | grep -v grep | wc -l)
+          if [ $IS_DATABASE_VALID -eq 0  ]; then
+            echo "[Error]: The provided database name is not accessible with the provided MySQL username and password. See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mysql-monitoring-integration#config for more info." >> /dev/stderr
             exit 2
```
Comment (jbeveland27) [inline recipes/newrelic/infrastructure/ohi/mysql/rhel.yml:102]: Should the exit code be incremented to `3` to be able to distinguish the type of error?

File: recipes/newrelic/infrastructure/ohi/mysql/suse.yml [hunk 3]
```diff
@@ -89,7 +89,13 @@ install:
...
-            echo -e "[Error]: Provided user has no access to the desired database.\n - See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mysql-monitoring-integration#config for more info."
+            echo "[Error]: Provided user has no access to the desired database. See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mysql-monitoring-integration#config for more info." >> /dev/stderr
+            exit 2
+          fi
+        - |
+          IS_DATABASE_VALID=$(mysql -u{{.NR_CLI_DB_USERNAME}} --port {{.NR_CLI_DB_PORT}} -p{{.NR_CLI_DB_PASSWORD}} -e "use {{.NR_CLI_DATABASE}};SELECT CURDATE()" 2>&1 | grep -v "mysql: [Warning] Using a password on the command line interface can be insecure." | grep 'CURDATE()' | grep -v grep | wc -l)
+          if [ $IS_DATABASE_VALID -eq 0  ]; then
+            echo "[Error]: The provided database name is not accessible with the provided MySQL username and password. See https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list/mysql-monitoring-integration#config for more info." >> /dev/stderr
             exit 2
```
Comment (jbeveland27) [inline recipes/newrelic/infrastructure/ohi/mysql/suse.yml:99]: Should the exit code be incremented to `3` to be able to distinguish the type of error?

Comment (Julien4218) [reply recipes/newrelic/infrastructure/ohi/mysql/suse.yml:99]: I don't think so, it's still in the same step of validating the DB access. We've been pretty consistent on rc 2 for validating the inputs with the specific OHI, rc 1 been no infra-agent.

File: recipes/newrelic/infrastructure/ohi/mysql/debian.yml (hunk 1 above)
Comment (Julien4218) [reply recipes/newrelic/infrastructure/ohi/mysql/debian.yml:102]: I don't think so, it's still in the same step of validating the DB access. We've been pretty consistent on rc 2 for validating the inputs with the specific OHI, rc 1 been no infra-agent.

File: recipes/newrelic/infrastructure/ohi/mysql/rhel.yml (hunk 2 above)
Comment (Julien4218) [reply recipes/newrelic/infrastructure/ohi/mysql/rhel.yml:102]: I don't think so, it's still in the same step of validating the DB access. We've been pretty consistent on rc 2 for validating the inputs with the specific OHI, rc 1 been no infra-agent.

**Comment by Julien4218** [comment]
@zlesnr @jbeveland27 I've revised the recipe to handle blank database name. I've validated the behavior on the various OS we support.
Feel free to re-review.
@jbeveland27 regarding error code value, I think there is an interesting conversation we'd want to have to standardize our rc values. I know @ctrombley is interested in that as well. Let's talk about this independently of this work if you agree.

**Pull request merged**